### ISSUE TITLE
[CN] Unknown constant response from SMT solver

### ISSUE THREAD

**Issue by peterohanley** [rems-project/cerberus#505]
This was really hard get a small reproduction for and I still have no idea what's causing this. @yav seems to be an SMT integration issue.

```c
struct state {
    // they also need to be these types for some reason
    int need[3];
    int all[3];
    int of[3];
    char these[3];
};
void foo(struct state *state);
/*@ spec foo(pointer state);
  requires
  take statein = Owned(state);
  each(u64 j; j < (u64)3u8) {statein.these[j] < 3u8};
  ensures
  take stateout = Owned(state);
  @*/

static int bar(struct state *state);
/*@ spec bar(pointer state);
  requires
  true;
  take si = Owned(state);
  each(u64 j;0u64 <= j && j < (u64)3u8) { si.these[j] < 3u8};
  ensures
  true;
  take so = Owned(state);
  each(u64 j;0u64 <= j && j < (u64)3u8) { so.these[j] < 3u8};
  @*/
static int baz(struct state *state);
/*@ spec baz(pointer state);
  requires
  true;
  ensures
  take s_out = Owned(state);
  each(u64 i; 0u64 <= i && i < (u64)3u8) { s_out.these[i] < 3u8};
  @*/
static int qux(struct state *state) ;
/*@ spec qux(pointer state);
  requires
  true;
  ensures
  return <= 0i32;
  take so = Owned(state);
  @*/
void foo(struct state *state)
{
    bar(state);
    baz(state);
    qux(state);
}
```
```
cn: internal error, uncaught exception:
    UnexpectedSolverResponse((error "line 92 column 15: unknown constant k!39"))
    Raised at Cn__Simple_smt.ack_command in file "backend/cn/lib/simple_smt.ml", line 518, characters 11-47
    Called from Stdlib__List.iter in file "list.ml", line 110, characters 12-15
    Called from Cn__Solver.model_evaluator in file "backend/cn/lib/solver.ml", line 1136, characters 4-38
    Called from Cn__Solver.provable.model_from in file "backend/cn/lib/solver.ml", line 1186, characters 15-38
    Called from Cn__ResourceInference.General.qpredicate_request_aux.(fun) in file "backend/cn/lib/resourceInference.ml", line 337, characters 21-71
    Called from Cn__Typing.map_and_fold_resources_internal.(fun) in file "backend/cn/lib/typing.ml", line 656, characters 27-35
    Called from Stdlib__List.fold_right in file "list.ml", line 126, characters 16-37
    -----8<-------
```

**Comment by peterohanley** [comment]
Seems to be fixed by #502

**Comment by dc-mak** [comment]
Brilliant, and thanks for the repro! I was working on a different feature when I triggered this and had to debug it on that. I'll merge in the fix now.
